Microsoft.AVS.ElasticSanDatastores/Microsoft.AVS.ElasticSanDatastores.psm1

```diff
@@ -0,0 +1,62 @@
+<#
+    .SYNOPSIS
+     This function expands an ElasticSAN datastore capacity in AVS vCenter.
+
+    .PARAMETER DatastoreName
+     Name of the ElasticSAN datastore to be expanded.
+
+    .EXAMPLE
+     Expand-ElasticSanDatastore -ClusterName Cluster-1 -DatastoreName MyElasticSanDatastore
+
+    .INPUTS
+     vCenter cluster name, Name of the ElasticSAN datastore.
+
+    .OUTPUTS
+     None.
+#>
+function Expand-ElasticSanDatastore {
+    [CmdletBinding()]
+    [AVSAttribute(10, UpdatesSDDC = $false, AutomationOnly = $false)]
+    Param (
+        [Parameter(
+            Mandatory=$true,
+            HelpMessage = 'Name of the ElasticSAN based datastore as seen on vCenter')]
+        [ValidateNotNull()]
+        [String]
+        $DatastoreName
+    )
+
+    if (-not $DatastoreName) {
```

Comment:
Is there a way to also validate whether the datastore provided is only associated with an Elastic San volume (Eg. Checking the NAA ID of the device associated with the VMFS volume)?